docs/spec_draft.md

```diff
@@ -941,6 +942,63 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dynamic_slice
+
+### Semantics
+
+Extracts a subtensor of the shape in `slice_sizes` from the `operand` tensor
+starting from `start_indices` and produces a `result` tensor. More formally,
+`result[i0, ..., iR-1] = operand[j0, ..., jR-1]` where
+`jd = start_indices[d] + id`.
+
+### Inputs
+
+| Name            | Type                                                     |
+|-----------------|----------------------------------------------------------|
+| `operand`       | tensor of any supported type                             |
+| `start_indices` | variadic number of 0-dimensional tensors of integer type |
+| `slice_sizes`   | 1-dimensional tensor of type `si64`                      |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same element type.
+  * (C2) size(`start_indices`) $=$ size(`slice_sizes`) $=$ rank(`operand`).
+  * (C3) Each `start_indices` `dk` is in range [0, shape(`operand`)[`k`]) for
+    all `k` in range [0, rank(`operand`)).
+  * (C4) `slice_sizes[k]` is in range [0, shape(`operand`)[`k`]) for all `k` in
+    range [0, rank(`operand`)).
+  * (C5) 0 $\le$ `k`th `start_indices` + `slice_sizes[k]` $\le$
```

Comment:
drop (C5) and (C3). `start_indices` being dynamic we do not know statically what the values are.

The above clamp mentioned in the semantics section is to  ensure that the extracted slice is always in-bounds with respect to the operand.